index.rst

```diff
@@ -265,41 +439,502 @@ element storing the following fields:
+
+.. parsed-literal::
+
+  {
+    "name" : "GBLinear",
+    "num_boosting_round" : *integer*,
+    "gblinear_train_param" : GBLinearTrainParam_,
+    "model": GBLinearModel_,
+    "updater": LinearUpdater_
+  }
+
+The ``num_boosting_round`` field stores the number of boosting rounds performed.
+
+GBLinearTrainParam
+------------------
+This class is a subclass of ``dmlc::Parameter``.
+
+.. parsed-literal::
+
+  {
+    "updater" : *string*,
+    "tolerance" : *floating-point*
+  }
+
+The ``updater`` field is the name of the linear updater used for training. Its
+value must match that of ``updater`` in GBLinear_.
+
+The ``tolerance`` field is the threshold for early stopping, in which iterations
+were terminated if the largest weight updater is smaller than the threshold.
+Setting it to zero disables early stopping.
+
+Note. ``max_row_perbatch`` is omitted because it is deprecated.
+
+GBLinearModel
+-------------
+
 .. parsed-literal::
 
   {
     "model_param" : GBLinearModelParam_,
     "weight" : [ *array of floating-point* ]
   }
 
+The ``weight`` field stores the final coefficients of the combined linear model,
+after all boosting rounds. Currently, the linear booster does not store
+coefficients of individual boosting rounds.
+
 GBLinearModelParam
 ------------------
 This class is a subclass of ``dmlc::Parameter``.
 
 .. parsed-literal::
 
   {
-    "num_feature" : *64-bit integer*,
+    "num_feature" : *integer*,
     "num_output_group" : *integer*
   }
 
-PredictorParam
+The ``num_output_group`` is the size of prediction per instance. This value is
+set to 1 for all tasks except multi-class classification. For multi-class
+classification, ``num_output_group`` must be set to the number of classes.
+
+LinearUpdater
+-------------
+Currently, we may choose one of the three subclasses for the linear updater:
+
+* CoordinateUpdater_: corresponds to ``updater="coord_descent"``
+* GPUCoordinateUpdater_: corresponds to ``updater="gpu_coord_descent"``
+* ShotgunUpdater_: corresponds to ``updater="shotgun"``
+
+We can determine which subclass was used by looking at the ``name`` field
+of each subclass.
+
+SoftmaxMultiClassObj
+--------------------
+This class is a subclass of Objective_.
+
+.. parsed-literal::
+
+  {
+    "name" : "SoftmaxMultiClassObj",
+    "num_class" : *integer*,
+    "output_prob" : *boolean*
+  }
+
+The ``num_class`` field must have the same value as ``num_class`` in
+LearnerTrainParam_.
+
+The ``output_prob`` field determines whether the loss function should produce
+class index (``false``) or probability distribution (``true``).
+
+HingeObj
+--------
+This class is a subclass of Objective_.
+
+.. parsed-literal::
+
+  {
+    "name" : "HingeObj"
+  }
+
+RegLossObj
+----------
+This class is a subclass of Objective_.
+
+.. parsed-literal::
+
+  {
+    "name" : "RegLossObj",
+    "loss_type" : *string*,
+    "scale_pos_weight": *floating-point*
+  }
+
+The ``loss_type`` field must be one of the following: ``LinearSquareLoss``,
```

Comment:
Did we update this in [#4267](https://github.com/dmlc/xgboost/pull/4267/files)? If not I still don't see why this would be called "Linear"